### PR TITLE
feat: share wallet details screen for shared wallet

### PR DESCRIPTION
**Pull request by shawnbusuttil**
# Checklist

- [ ] JIRA - \<LW-10550>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

image

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/create-flow/SharedWalletCreationStore.tsx [hunk 1]
```diff
@@ -174,17 +195,18 @@ type SharedWalletCreationStoreProps = {
...
+
+  const initialState = makeInitialState(activeWalletName || '');
+
   const [state, dispatch] = useReducer((prevState: State, action: Action): State => {
-    const stateMachine = makeStateMachine({
-      navigateHome: () => {
-        history.push(walletRoutePaths.sharedWallet.root);
-      }
-    });
+    const stateMachine = makeStateMachine({ navigateHome: setBackgroundPage });
```
[the remark this answers is not in the document]
Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/create-flow/SharedWalletCreationStore.tsx:209]: Why do we want a back to home?

File: apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/create-flow/SharedWalletCreationStore.tsx [hunk 2]
```diff
@@ -165,6 +171,21 @@ const makeStateMachine = ({ navigateHome }: { navigateHome: () => void }): State
...
+    return prevState;
+  },
+  [SharedWalletCreationStep.ShareDetails]: (prevState, action) => {
+    if (action.type === SharedWalletActionType.BACK) {
+      return {
+        ...prevState,
+        quorumRules: undefined,
+        step: SharedWalletCreationStep.Quorum
+      };
+    }
```
[the remark this answers is not in the document]
Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/create-flow/SharedWalletCreationStore.tsx:183]: Oops. You're right.

File: packages/core/src/ui/components/SharedWallet/ShareWalletDetails/ShareWalletDetails.module.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  &:hover {
+    background: none !important;
+  }
+}
+
+.icon {
+  border: none !important;
+  border-radius: 0 !important;
+  background: none !important;
+  margin-right: 40px !important;
```
[the remark this answers is not in the document]
Comment (shawnbusuttil) [reply packages/core/src/ui/components/SharedWallet/ShareWalletDetails/ShareWalletDetails.module.scss:18]: I tried without them already. It's unfortunate. Will review to possibly separate this in future.

File: packages/ui/src/design-system/action-card/action-card.component.tsx [hunk 4]
```diff
@@ -21,9 +23,9 @@ export const ActionCard = ({
   icon,
   ...props
 }: Readonly<Props>): JSX.Element => (
-  <Box {...props} className={cx.root}>
+  <Box {...props} className={classNames([cx.root, props.rootClassName])}>
```
[the remark this answers is not in the document]
Comment (shawnbusuttil) [reply packages/ui/src/design-system/action-card/action-card.component.tsx:26]: Yes I was making sure the important! was really needed here. :D

File: packages/ui/src/design-system/action-card/action-card.component.tsx [hunk 5]
```diff
@@ -6,13 +6,15 @@ import { Flex } from '../flex';
 import { Text } from '../text';
 
 import * as cx from './action-card.css';
-
+import classNames from 'classnames';
 import type { OmitClassName } from '../../types';
 
 type Props = Omit<OmitClassName<'div'>, 'title'> & {
-  title: { text: string; highlight: boolean }[];
+  title: { text: string; highlight?: boolean; bold?: boolean }[];
```
[the remark this answers is not in the document]
Comment (shawnbusuttil) [reply packages/ui/src/design-system/action-card/action-card.component.tsx:13]: I was going to but is it really worth trying to export the type safety of the weights from the UI for this one case?

[the remark this answers is not in the document]
Comment (szymonmaslowski) [reply packages/ui/src/design-system/action-card/action-card.component.tsx:13]: This is already located in the UI, so isn't it a matter of importing a proper type?

**Review by szymonmaslowski — APPROVED** [review]

**Pull request merged**